migration.md

```diff
@@ -1,58 +1,191 @@
+}
+```
+2. Now `Defaults[.period]` should be readable.
+
+---
+
 ### From `Codable Array/Dictionary` to `Native Array/Dictionary/Set`(With Native Supported Elements)
 
-Before v4.X, `Defaults` will store array/dictionary as a json string.  
-After v5.X, `Defaults` will store it as a native array/dictionary with native supported elements.
+Before v4.X, `Defaults` will store array/dictionary as a json string(`["a", "b", "c"]`).  
+
+After v5.X, `Defaults` will store it as a native array/dictionary with native supported elements(`[a, b, c]` ). 
 
-Before migration, your code should be like this
+#### Before migration, your code should be like this
 ```swift
 extension Defaults.Keys {
 	static let arrayString = Defaults.Key<[String]?>("arrayString")
 	static let dictionaryStringInt = Defaults.Key<[String: Int]?>("dictionaryStringInt")
+	static let dictionaryStringIntInArray = Defaults.Key<[[String: Int]]?>("dictionaryStringIntInArray")
 }
 ```
 #### Migration steps
-1. Call `Defaults.migration(.arrayString)`, `Defaults.migration(.dictionaryStringInt)`.
-2. Now `Defaults[.arrayString]`, `Defaults[.dictionaryStringInt]` should be readable.
+1. Call `Defaults.migration(.arrayString)`, `Defaults.migration(.dictionaryStringInt)`, `Defaults.migration(.dictionaryStringIntInArray)`.
+2. Now `Defaults[.arrayString]`, `Defaults[.dictionaryStringInt]`, `Defaults[.dictionaryStringIntInArray]` should be readable.
 
 ---
 
 ### From `Codable Array/Dictionary` to `Native Array/Dictionary/Set`(With Codable Elements)
 
-Before v4.X, `Defaults` will store array/dictionary as a json string.  
-After v5.X, `Defaults` will store it as a native array/dictionary with codable elements.
+Before v4.X, `Defaults` will store array/dictionary as a json string(`["{ "id": "0", "name": "Asia/Taipei" }"]`, `"10 Minutes"`).    
 
-Before migration, your code should be like this
+After v5.X, `Defaults` will store it as a native array/dictionary with codable elements(`[{ "id": "0", "name": "Asia/Taipei" }]`, `10 Minutes`).
+
+#### Before migration, your code should be like this
 ```swift
 private struct TimeZone: Codable {
 	var id: String
 	var name: String
 }
+private enum Period: String, Codable {
+	case tenMinutes = "10 Minutes"
+	case halfHour = "30 Minutes"
+	case oneHour = "1 Hour"
+}
 
 extension Defaults.Keys {
 	static let arrayTimezone = Defaults.Key<[TimeZone]?>("arrayTimezone")
+	static let arrayPeriod = Defaults.Key<[Period]?>("arrayPeriod")
 	static let dictionaryTimezone = Defaults.Key<[String: TimeZone]?>("dictionaryTimezone")
+	static let dictionaryPeriod = Defaults.Key<[String: Period]?>("dictionaryPeriod")
 }
 ```
 #### Migration steps
-1. Let `TimeZone` protocol conform to `Defaults.Serializable`
+1. Let `TimeZone` and `Period` protocol conform to `Defaults.Serializable`
 ```swift
 private struct TimeZone: Defaults.Serializable & Codable {
 	var id: String
 	var name: String
 }
+
+private enum Period: String, Defaults.Serializable & Codable {
```

Comment:
You use `Defaults.Serializable & Codable` here while you use `Defaults.Serializable, Codable` in other places. It should be consistent.